### DISCUSSION TITLE
Is the main-snapshot version recommended for ruleset maintainers?

### DISCUSSION THREAD

**Discussion by shuhuang-faire** [detekt/detekt#7601] [General]
We need to upgrade to kotlin 2.0.0. As advised we are trying to use the main-SNAPSHOT version. However the upgrade breaks a lot of custom rules written for 1.23.6. Do you suggest us to use the snapshot version and fix rules or wait until detekt 2.0 is officially released? Is there a timeline? 
Thanks!

**Comment by BraisGabin** [comment]
> As advised we are trying to use the main-SNAPSHOT version.

Where is that advised? You should be able to use the last stable version with kotlin 2.0.x. I use the last stable with kotlin2.

We want as many users on the snapshot as possible to provide feedback but we don't ensure any type of stability and we don't have any migration guide yet. So you can take the decission.
